server/mdm/apple/util.go

```diff
@@ -0,0 +1,71 @@
+package apple_mdm
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+)
+
+// Note Apple rejects CSRs if the key size is not 2048.
+const rsaKeySize = 2048
+
+// NewPrivateKey creates an RSA private key
+func NewPrivateKey() (*rsa.PrivateKey, error) {
```

Comment:
Looks like this (and a few others in this file) could be unexported. Although it looks like this could be a more general "crypto" package, as I don't think there's anything apple-specific (except maybe the RSA key size requirement). Definitely not a blocker for the PR at this point, we can refactor this when it starts getting used or required elsewhere.

Reply:
99217ebfe